src/phoebus_driver.cpp

```diff
@@ -1014,11 +1015,41 @@ TaskListStatus PhoebusDriver::MonteCarloStep() {
 
 /**
  * Gets called before output.
- * Currently: Fills tracers.
+ * Currently:
+ *  Fills Tracers
+ *  Computes entropy for output
  **/
 void UserWorkBeforeOutput(MeshBlock *pmb, ParameterInput *pin) {
   auto tracer_pkg = pmb->packages.Get("tracers");
   bool do_tracers = tracer_pkg->Param<bool>("active");
+
+  namespace p = fluid_prim;
+  auto rc = pmb->meshblock_data.Get().get();
+  PackIndexMap imap;
+  auto v = rc->PackVariables(
+      {p::density::name(), p::ye::name(), p::temperature::name(), p::entropy::name()},
+      imap);
+
+  const int irho = imap[p::density::name()].first;
+  const int iye = imap[p::ye::name()].first;
+  const int itemp = imap[p::temperature::name()].first;
+  const int is = imap[p::entropy::name()].first;
+
+  IndexRange ib = rc->GetBoundsI(IndexDomain::interior);
+  IndexRange jb = rc->GetBoundsJ(IndexDomain::interior);
+  IndexRange kb = rc->GetBoundsK(IndexDomain::interior);
+
+  auto eos = pmb->packages.Get("eos")->Param<Microphysics::EOS::EOS>("d.EOS");
+  parthenon::par_for(
+      DEFAULT_LOOP_PATTERN, "UserWorkBeforeOutput::H5", DevExecSpace(), kb.s, kb.e, jb.s,
+      jb.e, ib.s, ib.e, KOKKOS_LAMBDA(const int k, const int j, const int i) {
+        Real lambda[2];
+        lambda[0] = v(iye, k, j, i);
```

Comment:
Ah, got it. Since we are supporting any EOS, we have to be more careful with `ye`. In general, it may not be enabled (such as the crashing bondi test). Have a look, e.g., [here](https://github.com/lanl/phoebus/blob/5bc3b2eb221d276ccb83071fa76a6e1449833438/src/pgen/advection.cpp#L80) @mari2895 for how to handle non-enabled ye cases.

Reply:
@AstroBarker Thanks!

Reply:
added the check